Source/engine/path.cpp

```diff
@@ -162,7 +162,7 @@ int ReconstructPath(const ExploredNodes &explored, PointT dest, int8_t *path, si
 		if (it->second.g == 0) break; // reached start
 		path[len++] = GetPathDirection(it->second.prev, cur);
 		cur = it->second.prev;
-		if (len == maxPathLength) {
+		if (len > maxPathLength) {

```

Comment:
`len` here is the number of steps rather than positions, so if it's 1 that's exactly 1 step 🤔

Reply:
Ah, nvm, in this branch we set length to 0.